SharpPcap/DeviceConfiguration.cs

```diff
@@ -44,6 +44,23 @@ public class DeviceConfiguration
 
         public bool? Immediate { get; set; }
 
+        #region File IO
+        /// <summary>
+        /// Reading and writing capture files
+        /// </summary>
+        public string Filename { get; set; }
+
+        /// <summary>
+        /// Writing capture files
+        /// </summary>
+        public System.IO.FileMode Filemode { get; set; }
+
+        /// <summary>
+        /// Writing capture files
+        /// </summary>
+        public PacketDotNet.LinkLayers LinkLayerType { get; set; }
```

Comment:
This could prove interesting when combined with `pcap_set_datalink`, also make it nullable, since default value 0 is a valid link type.
```suggestion
        public PacketDotNet.LinkLayers? LinkLayerType { get; set; }
```